primitives/core/src/ecdsa.rs

```diff
@@ -190,6 +190,21 @@ impl rstd::convert::TryFrom<&[u8]> for Signature {
 	}
 }
 
+#[cfg(feature = "std")]
+impl Serialize for Signature {
+	fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error> where S: Serializer {
+		serializer.serialize_str(&hex::encode(self))
+	}
+}
+
+#[cfg(feature = "std")]
+impl<'de> Deserialize<'de> for Signature {
+	fn deserialize<D>(deserializer: D) -> Result<Self, D::Error> where D: Deserializer<'de> {
+		Ok(Signature::from_slice(&hex::decode(&String::deserialize(deserializer)?)
```

Comment:
That will crash the node if we ever attempt to deserialize a wrongly sized signature. Please use `try_from` instead and return an error instead of panicking.

Reply:
Resolved in 4e2c7f0

Reply:
Resolved in 4e2c7f0